pandas/tests/io/formats/test_to_csv.py

```diff
@@ -285,3 +286,21 @@ def test_to_csv_string_array_utf8(self):
             df.to_csv(path, encoding='utf-8')
             with open(path, 'r') as f:
                 assert f.read() == expected_utf8
+
+    def test_to_csv_stdout_file(self):
+        # GH 21561
+        str_array = [{'names': ['foo', 'bar']}, {'names': ['baz', 'qux']}]
+        df = pd.DataFrame(str_array)
```

Comment:
might be simpler to:
`DataFrame([['foo', 'bar'], ['baz', 'qux']], columns=['name_1', 'name_2'])`